### ISSUE TITLE
generate paper-compatible task function dataset

### ISSUE THREAD

**Issue by KiaraGrouwstra** [tycho01/synthesis#2]
~~note that I should incorporate their task functions (test set), building blocks, and grammar (I wanna keep mine simple!).~~

**Comment by KiaraGrouwstra** [comment]
I implemented Tamandu under the impression it was the only one not requiring recursion, then realized that ~~(1) I was wrong, and (2)~~ its polymorphism lacks any variation, i.e. for my purposes doesn't count as proper type polymorphism, meaning my intended approach is logically unlikely to add significant value on this benchmark.

This doesn't necessarily mean I can't just go ahead and proceed with other stuff -- it just means I may need to settle on a nicer dataset at some point.
In a sense I feel a bit like putting this off until maybe further literature research -- ~~I'm not so happy about the recursion (#12) required by the papers I was currently considering, so I'm kinda hoping to still somehow find something better there.~~

For what it's worth, Tamandu should do fine for the purpose of serving as a dataset to sorta check if my [NSPS](https://arxiv.org/abs/1611.01855) implementation ends up sane.

I'll downgrade the priority of this issue to reflect these advances.

**Comment by KiaraGrouwstra** [comment]
I'd like to narrow the scope of this issue.
my goal here is not yet to construct the perfect final dataset to show the strength of my approach (which I'll push to #9), but rather to get something so as to just unblock the other tasks like building the synthesizers, even if the initial dataset might not initially be designed to demonstrate the strengths of my approach yet.

**Comment by KiaraGrouwstra** [comment]
progress update: while `succ` now works, after better handling a [*bunch*](https://github.com/tycho01/synthesis/commit/8c40461029acba53744b470484bc8b5006c66a37), the majority of the blocks still result in run-time issues.

**Comment by KiaraGrouwstra** [comment]
at this point this seems to just depend on combinators (#4). closing in favor of that issue.
